From 491de9436c0f7e7615ac020dafcedae6731a2163 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:15 -0400
Subject: [tools] perf: Do not auto-remove Children column if --fields given

Message-id: <1410181752-25631-251-git-send-email-jolsa@redhat.com>
Patchwork-id: 91987
O-Subject: [PATCH RHEL7.1 BZ1134356 250/307] perf tools: Do not auto-remove Children column if --fields given
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 2bf1a12340bda1bf621f27b9892094a51b1297fd
Author: Namhyung Kim <namhyung@kernel.org>
Date: Thu Mar 20 09:10:29 2014 +0900

description
===========
Depending on the configuration perf inserts/removes the Children
column in the output automatically.  But it might not be what user
wants if [s]he give --fields option explicitly.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/ui/hist.c b/tools/perf/ui/hist.c
index 8ca6387..498adb2 100644
--- a/tools/perf/ui/hist.c
+++ b/tools/perf/ui/hist.c
@@ -513,6 +513,9 @@ void perf_hpp__column_disable(unsigned col)
 
 void perf_hpp__cancel_cumulate(void)
 {
+ if (field_order)
+  return;
+
  perf_hpp__column_disable(PERF_HPP__OVERHEAD_ACC);
  perf_hpp__format[PERF_HPP__OVERHEAD].header = hpp__header_overhead;
 }
-- 
1.7.1